[Term Entry] Python:Pandas built-in-functions: .concat() (#4814)
Codecademy/docs: content/pandas/concepts/built-in-functions/terms/concat/concat.md

---
Title: '.concat()'
Description: 'Concatenates multiple Dataframes or Series along a particular axis'
Subjects: # Please only use Subjects in the subjects.md file (https://github.com/Codecademy/docs/blob/main/documentation/subjects.md). If that list feels insufficient, feel free to create a new Subject and add it to subjects.md in your PR!
  - 'A subject name'
  - 'A second subject name'
  - 'An nth subject name'
Tags: # Please only use Tags in the tags.md file (https://github.com/Codecademy/docs/blob/main/documentation/tags.md). If that list feels insufficient, feel free to create a new Tag and add it to tags.md in your PR!
  - 'Data Structures'
  - 'Functions'
  - 'Pandas'
  - 'CSV'
  - 'Data'
CatalogContent: # Please use course/path landing page slugs, rather than linking to individual content items. If listing multiple items, please put the most relevant one first
  - 'learn-Data-Analysis-with-Pandas'
  - 'paths/data-science'

> ---

Reviewer: added relevant subjects and tags from the documents

> The `.concat()` function is used to concatenate and combine multiple [`DataFrames`](https://www.codecademy.com/resources/docs/pandas/dataframe) or [`Series`] along a particular axis.

Reviewer: ```suggestion
The **`.concat()`** function is used to concatenate and combine multiple [`DataFrames`](https://www.codecademy.com/resources/docs/pandas/dataframe) or [`Series`] along a specified axis.
```

Reviewer: The first instance of the term should be wrapped in bold

## Syntax

```pseudo

> pandas.concat(objs)

Reviewer: The syntax should include all the parameters from the documentation of the `.concat()` method. I found out the following two:

`pandas.concat(objs, axis=0, join='outer', ignore_index=False, keys=None, levels=None, names=None, verify_integrity=False, sort=False, copy=True)`

`pandas.concat(objs, axis, join, ignore_index, keys, levels, names, verify_integrity, sort, copy)`

You can use either of them

|   `ignore_index`   |                bool, default `False`                 | If `True`, it resets the index in the resulting DataFrame                              |
|       `keys`       |        sequence (list or tuple), default None        | Lets you construct a hierarchical index using the provided keys as the outermost level |
|      `levels`      |           list of sequences, default None            | Specific levels to use for constructing a MultiIndex if keys are provided              |
|      `names`       |              list of str, default None               | Names for the levels generated in the hierarchical index                               |
| `verify_integrity` |                bool, default `False`                 | If `True`, checks whether the new concatenated axis contains duplicates                |
|       `sort`       |                bool, default `False`                 | If `True`, it sorts the resulting Series or Dataframe by the keys                      |
|       `copy`       |                 bool, default `True`                 | If `False`, it avoids copying data unnecessarily                                       |

> 

Reviewer: All the parameters need to be described in the below format:

- `objs`: A sequence or mapping of `Series` or `DataFrame` objects. Can be a list of objects or a dictionary where the keys will be used for the `keys` parameter.
- `axis`: Specifies the axis along which to concatenate the objects. The default value is 0.
And so on, so in this format describe the other parameters as well.

## Example

> 

Reviewer: ```suggestion

The example below demonstrates the use of `.concat()` method:

```

```py

> import pandas

Reviewer: ```suggestion
import pandas as pd
```

Reviewer: since you are using the abbreviation pd in the code

df1 = pd.DataFrame({'A': [1, 2], 'B': [3, 4]})
df2 = pd.DataFrame({'A': [5, 6], 'B': [7, 8]})

result = pd.concat([df1, df2])
print(result)

```

> The example will result in a new DataFrame is returned by concatenating df1 and df2 along the rows:

Reviewer: ```suggestion
The example will result in a new `DataFrame` created by concatenating `df1` and `df2` along the rows. The output is as follows:
```

```shell
   A   B
0  1   2
1  3   4
0  5   6

> 1  7   8

Reviewer: The output that I'm getting is somewhat different, could you please run the code and check again? The is the output I'm getting:
```
   A  B
0  1  3
1  2  4
0  5  7
1  6  8
```

---
Title: '.concat()'

> Description: 'Concatenates multiple Dataframes or Series along a particular axis.'

Reviewer 2: ```suggestion
Description: 'Concatenates multiple dataframes or series along a particular axis.'
```

Subjects:
  - 'Computer Science'
  - 'Data Science'
Tags:
  - 'Data Structures'
  - 'Functions'
  - 'Pandas'
CatalogContent:

>   - 'paths/data-science'
>   - 'learn-python-3'
>   - 'paths/computer-science'
>   - 'paths/data-science-foundations'

Reviewer 2: ```suggestion
  - 'paths/data-science'
  - 'learn-python-3'
  - 'paths/computer-science'
  - 'paths/data-science-foundations'
```
We prefer to keep 2 CatalogContent here:
- Free course
- Pro course/path
Can you modify it accordingly?

---

The **`.concat()`** function is used to concatenate and combine multiple [`DataFrames`](https://www.codecademy.com/resources/docs/pandas/dataframe) or `Series` along a particular axis.

## Syntax

```pseudo
pandas.concat(objs, axis=0, join='outer', ignore_index=False, keys=None, levels=None, names=None, verify_integrity=False, sort=False, copy=True)
```

- `objs`: Denotes the objects to concatenate, which can be a sequence or mapping of pandas `Series` or `DataFrame` objects. It must be specified and can be passed as a list, tuple, dictionary, Series, or DataFrame.
- `axis`: Specifies the axis along which to concatenate the objects. The default value is 0 for rows, while 1 represents columns.

> - `join`: Determines how to handle indexes on other axes. Options include "outer" (default), "inner," "left," or "right."

Reviewer 2: Can you recheck if the `join` parameter supports "left" or "right" options?

> - `objs`: Denotes the objects to concatenate, which can be a sequence or mapping of pandas `Series` or `DataFrame` objects. It must be specified and can be passed as a list, tuple, dictionary, Series, or DataFrame.
> - `axis`: Specifies the axis along which to concatenate the objects. The default value is 0 for rows, while 1 represents columns.
> - `join`: Determines how to handle indexes on other axes. Options include "outer" (default), "inner," "left," or "right."

Reviewer 2: ```suggestion
- `objs`: Denotes the objects to concatenate, which can be a sequence or mapping of pandas `Series` or `DataFrame` objects. It must be specified and can be passed as a list, tuple, dictionary, Series, or DataFrame.
- `axis`: Specifies the axis to concatenate the objects. The default value is 0 for rows, while 1 represents columns.
- `join`: Determines how to handle indexes on other axes. Options include "outer" (default), "inner," "left," or "right."
```

- `ignore_index`: If `True`, resets the index in the resulting DataFrame. The new axis will be labeled 0, ..., n-1. Default value is `False`.
- `keys`: Constructs a hierarchical index using the provided keys as the outermost level. Default value is `None`.
- `levels`: Specific levels to use for constructing a MultiIndex if keys are provided. Default value is `None`.
- `names`: Names for the levels generated in the hierarchical index. Default value is `None`.
- `verify_integrity`: If `True`, checks whether the new concatenated axis contains duplicates. Default value is `False`.

> - `sort`: If `True`, sorts the resulting `Series` or `Dataframe` by the keys. Default value is `False`.

Reviewer 2: It sorts non-concatenation axis if it is not already aligned. Please recheck this too.

> - `ignore_index`: If `True`, resets the index in the resulting DataFrame. The new axis will be labeled 0, ..., n-1. Default value is `False`.
> - `keys`: Constructs a hierarchical index using the provided keys as the outermost level. Default value is `None`.
> - `levels`: Specific levels to use for constructing a MultiIndex if keys are provided. Default value is `None`.
> - `names`: Names for the levels generated in the hierarchical index. Default value is `None`.
> - `verify_integrity`: If `True`, checks whether the new concatenated axis contains duplicates. Default value is `False`.
> - `sort`: If `True`, sorts the resulting `Series` or `Dataframe` by the keys. Default value is `False`.

Reviewer 2: ```suggestion
- `ignore_index`: If `True`, reset the index in the resulting DataFrame. The new axis will be labeled 0, ..., n-1. The default value is `False`.
- `keys`: Constructs a hierarchical index using the provided keys as the outermost level. The default value is `None`.
- `levels`: Specific levels to use for constructing a MultiIndex if keys are provided. The default value is `None`.
- `names`: Names for the levels generated in the hierarchical index. The default value is `None`.
- `verify_integrity`: If `True`, checks whether the new concatenated axis contains duplicates. The default value is `False`.
- `sort`: If `True`, sorts the resulting `Series` or `Dataframe` by the keys. The default value is `False`.
```

> - `copy`: If `False`, avoids copying data unnecessarily. Default value is `True`.

Reviewer 2: ```suggestion
- `copy`: If `False`, avoid copying data unnecessarily. The default value is `True`.
```

> **Note:** Only the `objs` parameter is required; all other parameters are optional.

## Example

The example below demonstrates the use of `.concat()` method:

```py
import pandas as pd

df1 = pd.DataFrame({'A': [1, 2], 'B': [3, 4]})
df2 = pd.DataFrame({'A': [5, 6], 'B': [7, 8]})

result = pd.concat([df1, df2])
print(result)
```

The example will result in a new `DataFrame` created by concatenating `df1` and `df2` along the rows. The output is as follows:

```shell
   A   B
0  1   3
1  2   4
0  5   7
1  6   8
```

## Codebyte Example

The code demonstrates the `.concat()` function on two DataFrames, concatenating `df1` and `df2` column-wise (`axis=1`) and using `keys` to create a hierarchical column index:

```codebyte/python
import pandas as pd

df1 = pd.DataFrame({'A' : [1,2,3,4,5], 'B' : [6,7,8,9,10]})
df2 = pd.DataFrame({'C' : [11,12,13,14,15], 'D' : [16,17,18,19,20]})

result = pd.concat([df1, df2], axis=1, keys = ['df1', 'df2'])

print(result)

> 

Reviewer 2: ```suggestion
```
Unnecessary space.
